zvariant/src/type.rs

```diff
@@ -77,6 +77,87 @@ pub trait DynamicDeserialize<'de>: DynamicType {
         S::Error: Into<zvariant::Error>;
 }
 
+macro_rules! impl_type {
```

Comment:
Nice macro! I especially love how it also handles tests. :+1: Having said that, it would be nice to:

* Add a doc comment explaining the usage. It doesn't have to be detailed.
* rename to be a bit more descriptive. `impl_type_for_repr`? Better names welcome of course. :)

Reply:
Agreed, the macro name could be a bit more descriptive. How about `impl_type_with_repr`? I'm by no means happy with this name, but I think it's slightly better than `impl_type_for_repr`?

Reply:
sure.